utils/rjson.hh

```diff
@@ -393,6 +394,26 @@ public:
     bool EndObject(rapidjson::SizeType memberCount = 0) { return _writer.EndObject(memberCount); }
     bool StartArray() { return _writer.StartArray(); }
     bool EndArray(rapidjson::SizeType elementCount = 0) { return _writer.EndArray(elementCount); }
+
+    template<typename U>
+    bool Write(U v) {
+        using T = std::remove_cvref_t<U>;
+        if constexpr (std::same_as<T, bool>) {
+            return Bool(v);
+        } else if constexpr (std::same_as<T, int>) {
+            return Int(v);
+        } else if constexpr (std::same_as<T, unsigned>) {
+            return Uint(v);
+        } else if constexpr (std::same_as<T, int64_t>) {
+            return Int64(v);
+        } else if constexpr (std::same_as<T, uint64_t>) {
+            return Uint64(v);
+        } else if constexpr (std::same_as<T, double>) {
+            return Double(v);
+        } else if constexpr (std::convertible_to<T, std::string_view>) {
+            return String(v);
+        }
```

Comment:
We need an else clause with an error. Otherwise a type which doesn't match any of the above, will fall through and be silently ignored.

Reply:
no, it will not be silently ignored if an unknown type is passed in. as the compiler would warn and error out if a function claims to return a `bool` but it fails to do so in any of its branches.

Reply:
Ah, I see. Sneaky, but it works.

Reply:
An `else` branch would still serve to document this, e.g.:
```c++
} else {
    // This intentionally breaks compilation if U doesn't match any of the above types.
    return void;
}
```

But I won't insist.

Reply:
For me the important thing is that compilation does break and the developer is forced to investigate. If we can help them with hints, it is good, but I won't block this PR on that.

Reply:
i see. so the problem is already addressed. let's make it a bonus for the follow-up then.

Reply:
bonus taken!